libs/environment/src/components/network-loader/network-loader.tsx

```diff
@@ -0,0 +1,163 @@
+import { useState, useEffect, useMemo } from 'react';
+import type { ReactNode, FC } from 'react';
+import type { ApolloClient } from '@apollo/client';
+import { ApolloProvider } from '@apollo/client';
+import {
+  Callout,
+  Intent,
+  Button,
+  Icon,
+  Loader,
+} from '@vegaprotocol/ui-toolkit';
+import { t } from '@vegaprotocol/react-helpers';
+import { useEnvironment } from '../../hooks';
+import type { ConfigStatus } from '../../types';
+
+type NetworkLoaderProps<T> = {
+  children?: ReactNode;
+  skeleton?: ReactNode;
+  createClient: (url: string) => ApolloClient<T>;
+};
+
+type StatusComponentProps = {
+  status: ConfigStatus;
+  children?: ReactNode;
+};
+
+type MessageComponentProps = {
+  children: ReactNode;
+};
+
+type ErrorComponentProps = MessageComponentProps & {
+  showTryAgain?: boolean;
+};
+
+const Error: FC<ErrorComponentProps> = ({
+  children,
+  showTryAgain,
+}: ErrorComponentProps) => (
+  <div>
+    <div className="mb-16">{children}</div>
+    {showTryAgain && (
+      <Button
+        className="mt-8"
+        variant="secondary"
+        onClick={() => window.location.reload()}
```

Comment:
question (blocking): Will this error in our next app? We might need to check for window first `typeof window !== 'undefined'`

Reply:
makes sense!